packages/sources/dlc-btc-por/src/endpoint/proof-of-reserves.ts

```diff
@@ -1,16 +1,33 @@
 import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
-import { EmptyInputParameters } from '@chainlink/external-adapter-framework/validation/input-params'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation/input-params'
 import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
 import { config } from '../config'
 import { porTransport } from '../transport/proof-of-reserves'
 
+export const inputParameters = new InputParameters(
```

Comment:
more of a philosophical comment, but rather than each NOP setting the `{NETWORK}_DLC_CONTRACT`, which would be the same across all NOPs, I'd lean towards adding the contract address as an input param. Useful in the unlikely event of a contract change.

Not sure if it's entirely safe to generalize, but let's restrict env vars here to individual configs (such as RPC URLs) that differ between NOPs, and use input params for configs across all NOPs.

Reply:
Agree, on the other hand this will make even more confusion between legacy env vars and new input params. I'm kinda thinking now that it worth just removing old env vars, make all those breaking changes and ensure NOPs configure everything properly before running new images. Otherwise there will be very strange behavior which can lead to errors. wdyt ?